docs/protocols/core-concepts.mdx

```diff
@@ -343,197 +324,139 @@ await a2a.send({
   "task_type": "create_media_buy",
   "status": "completed",
   "timestamp": "2025-01-22T10:30:00Z",
+  "message": "Media buy created successfully",
   "result": {
     "media_buy_id": "mb_12345",
     "buyer_ref": "nike_q1_campaign_2024",
     "creative_deadline": "2024-01-30T23:59:59Z",
     "packages": [
-      {
-        "package_id": "pkg_12345_001",
-        "buyer_ref": "nike_ctv_sports_package"
-      },
-      {
-        "package_id": "pkg_12345_002",
-        "buyer_ref": "nike_audio_drive_package"
-      }
+      { "package_id": "pkg_12345_001", "buyer_ref": "nike_ctv_sports_package" }
     ]
   }
 }
 ```
 
-#### For Other Async Operations
+#### A2A Webhook POST
 
-Each async operation posts its specific response schema:
+A2A sends `Task` (for final states) or `TaskStatusUpdateEvent` (for progress updates):
 
-- **`activate_signal`** → `activate-signal-response.json`
-- **`sync_creatives`** → `sync-creatives-response.json`
-- **`update_media_buy`** → `update-media-buy-response.json`
+```json
+{
+  "id": "task_456",
+  "contextId": "ctx_123",
+  "status": {
+    "state": "completed",
+    "message": {
+      "role": "agent",
+      "parts": [
+        { "text": "Media buy created successfully" },
+        {
+          "data": {
+            "media_buy_id": "mb_12345",
+            "buyer_ref": "nike_q1_campaign_2024",
+            "creative_deadline": "2024-01-30T23:59:59Z",
+            "packages": [
+              { "package_id": "pkg_12345_001", "buyer_ref": "nike_ctv_sports_package" }
+            ]
+          }
+        }
+      ]
+    },
+    "timestamp": "2025-01-22T10:30:00Z"
+  }
+}
+```
 
-#### Webhook URL Patterns
+#### Status-Specific Data Schemas
 
-Structure your webhook URLs to identify the operation and agent:
+The data payload (`result` in MCP, `status.message.parts[].data` in A2A) uses status-specific schemas:
 
-```
-https://buyer.com/webhooks/adcp/{task_name}/{agent_id}/{operation_id}
-```
+| Status | Data Schema | Contents |
+|--------|-------------|----------|
+| `completed` | `[task]-response.json` | Full task response (success branch) |
+| `failed` | `[task]-response.json` | Full task response (error branch) |
+| `working` | `[task]-async-response-working.json` | Progress info (`percentage`, `step`) |
+| `input-required` | `[task]-async-response-input-required.json` | Requirements, approval data |
+| `submitted` | `[task]-async-response-submitted.json` | Acknowledgment (usually minimal) |
 
-**Example URLs:**
-- `https://buyer.com/webhooks/adcp/create_media_buy/agent_abc/op_xyz`
-- `https://buyer.com/webhooks/adcp/activate_signal/agent_abc/op_123`
-- `https://buyer.com/webhooks/adcp/sync_creatives/agent_abc/op_456`
+#### Supported Async Operations
 
-Your webhook handler can parse the URL path to route to the correct handler based on the task name.
+Each async operation uses its main response schema for `completed`/`failed` statuses:
 
-#### Webhook Payload Structure
+| Task | Response Schema |
```

Comment:
I think we are going to have HITL for get products too as there are cases where proposals need human engagement. I think we should add that while we're in here. Could reporting be webhook too? I know we have the separate hook on create media buy, but wondering about even a standard `get_media_buy_delivery` taking a while for some reason